Add support for spatial averaging parallelism via Dask

pochedls opened the pull request:
## Description

This PR is intended to fix #115. It simply updates the `dataarray.values` attribute directly (when applicable) to maintain compatibility with chunked dataarray data (for spatial averaging routines). 

- Closes #115 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected) [Not Applicable]

pochedls committed:
Proposed changes to address Issue #115

pochedls committed:
updated commit removing .load

pochedls committed:
add unit tests and comments

pochedls committed:
fix code style issue

tomvothecoder reviewed (changes requested):
Looks good overall, just a few suggestions. Thanks!

tomvothecoder committed:
Fix numpy and xarray+dask errors in `_swap_lon_axis()`

tomvothecoder committed:
Update `_swap_lon_axes()` logic

tomvothecoder committed:
Fix `_force_lon_linearity` tests not identical
- Add `with xr.set_options(keep_attrs=True)` to `_swap_lon_axes()` to fix tests
- Add tests for `_swap_lon_axes()` and chunking
- Update name of tests

tomvothecoder committed:
Update docstrings

tomvothecoder committed:
Update chunking for a test

tomvothecoder committed:
Add `.load()` to `_scale_domain_to_region()`

tomvothecoder committed:
Remove outdated comment with `.values` attr

tomvothecoder committed:
Update line spacing and fix comment

tomvothecoder committed:
Add test for incorrect `to` arg

tomvothecoder committed:
Change order of `result` and `expected` vars

pochedls reviewed (commented):
@tomvothecoder - This looks good to me. It is passing tests and giving the correct output now.

tomvothecoder committed:
Rename test

tomvothecoder committed:
Update comment and docstring
